cmd/cloud.go

```diff
@@ -343,20 +344,69 @@ func getCmdCloud(gs *state.GlobalState) *cobra.Command {
 	}
 
 	exampleText := getExampleText(gs, `
-  {{.}} cloud script.js`[1:])
+  # Authenticate with Grafana k6 Cloud
+  $ {{.}} cloud login
+
+  # Run a k6 script in the Grafana k6 cloud
+  $ {{.}} cloud run script.js
+
+  # Run a k6 archive in the Grafana k6 cloud
+  $ {{.}} cloud run archive.tar
+
+  # [deprecated] Run a k6 script in the Grafana k6 cloud
+  $ {{.}} cloud script.js
+
+  # [deprecated] Run a k6 archive in the Grafana k6 cloud
+  $ {{.}} cloud archive.tar`[1:])
 
 	cloudCmd := &cobra.Command{
 		Use:   "cloud",
 		Short: "Run a test on the cloud",
-		Long: `Run a test on the cloud.
+		Long: `[deprecation notice]
+The k6 team is in the process of modifying and deprecating the cloud command behavior. In the future, the "cloud"
+command will only display a help text, instead of running tests in the cloud.
+To run tests in the cloud, users are now invited to migrate to the "k6 cloud run" command instead.
+
+Run a test on the cloud.
 
 This will execute the test on the k6 cloud service. Use "k6 login cloud" to authenticate.`,
-		Example: exampleText,
-		Args:    exactArgsWithMsg(1, "arg should either be \"-\", if reading script from stdin, or a path to a script file"),
+		Args:    exactCloudArgs(),
 		PreRunE: c.preRun,
 		RunE:    c.run,
+		Example: exampleText,
 	}
+
+	// Register `k6 cloud` subcommands
+	cloudCmd.AddCommand(getCmdCloudRun(gs))
+	cloudCmd.AddCommand(getCmdCloudLogin(gs))
+
 	cloudCmd.Flags().SortFlags = false
 	cloudCmd.Flags().AddFlagSet(c.flagSet())
+
 	return cloudCmd
 }
+
+func exactCloudArgs() cobra.PositionalArgs {
+	return func(_ *cobra.Command, args []string) error {
+		if len(args) < 1 || len(args) > 2 {
```

Comment:
In such case, now the message is something like:

```
ERRO[0000] accepts 1 arg(s), received 0: arg should either be "-", if reading script from stdin, or a path to a script file 
```

So, I'm wondering, should we try to be somehow more explicit? Or just delegate to `-h`?

Reply:
Also, not sure if 2 args is technically correct, or it should be one command plus one arg, if so.

Reply:
I always found it a bit counterintuitive that `k6 run` and `k6 cloud` return an error when missing an argument, as opposed to displaying "usage help text" instead for instance.

However, I also notice that so far we don't seem to have explicitly documented the use of the `-` argument, besides that specific error message, to allow for piping the script/archive as content either. 

So I would be keen to modify the help text to mention the argument structure, and support for '-' explicitly, and show the help text indeed in that case 👍🏻

Reply:
For now I've simplified the logic, and made it so that `k6 cloud` (without argument) displays

```
ERRO[0000] the k6 cloud command accepts 1 argument consisting in either in a subcommand such as `run` or `cloud`, or the path to a script/archive, or the `-` symbol, received: 0 arguments instead
```

And so that `k6 cloud run` (without script argument) displays

```
 accepts 1 arg(s), received 0: the k6 cloud run command expects a single argument consisting in either a path to a script or archive file, or the "-" symbol indicating the script or archive should be read from stdin
 ```
 
 Let me know if that's good enough, I'll resolve in the meantime, but feel free to unresolve 🙇‍♂️